module_utils/certificate/providers/certmonger.py

```diff
@@ -0,0 +1,136 @@
+import os
+
+from io import StringIO
+
+# pylint: disable=no-name-in-module
+from ansible.module_utils.six.moves.configparser import ConfigParser
+
+from .base import CertificateRequestBaseProvider
+
+
+class CertificateRequestCertmongerProvider(CertificateRequestBaseProvider):
+    """Certmonger provider for certificate Linux System Role."""
+
+    def __init__(self, *args, **kwargs):
+        super(CertificateRequestCertmongerProvider, self).__init__(*args, **kwargs)
+
+        self.requests_dir = "/var/lib/certmonger/requests"  # TODO: Allow to override?
+        self._request_file_path = self._find_certmonger_request_file()
+        self._certmonger_metadata = self._read_certmonger_request_file()
+
+    def _find_certmonger_request_file(self):
+        """Search for certificate metadata in certmonger data directory.
+
+        Looks for the certificate path, file by file inside certmonger data
+        directory.
+
+        Only open files that are 50k or smaller to avoid getting stuck for
+        a long time.
+
+        If the file is found it's returned immediately. If not found returns None.
+        """
+        ls_results = os.listdir(self.requests_dir)
+        line_start = "cert_storage_location="
+        for fname in ls_results:
+            file_path = os.path.join(self.requests_dir, fname)
+            with open(file_path, "r") as file_:
+                for line in file_:
+                    # find the line with the certificate location
+                    if line.startswith(line_start):
+                        location = line.split("=")[1].strip()
+                        if location == self.certificate_file_path:
+                            return file_path
+                        # not the right file, move to next file
+                        break
+        return None
```

Comment:
Is this expected to run as root? certmonger is a dbus app so you can query for things over that rather than doing filesystem I/O.

Reply:
I decided to go for filesystem because I could easily parse the file using configparser which is a python builtin and to use dbus I'd need to add the bindings as dependency.

By default dbus wouldn't also require root access? Also if I run getcert without root access I would get a msg 'Insufficient access.  Please retry operation as root.'

Reply:
Without knowing too much about certmonger, this sounds like a good point - if certmonger has a dbus interface, why doing things behind its back instead of asking it over dbus? The network role also communicates with NetworkManager over dbus.

Reply:
well... reading a ini-like file seemed a lot more straightforward to me since we are performing only reads in the files. Also by reading the file it prevent a dependency. 

@rcritten what risks do you see in the file implementation? What would be the benefits on using dbus for reading the CSR (pem data) and the ca name from dbus instead of doing from the file?

Reply:
There could be a race condition between the state inside certmonger and the state on disk. This isn't theoretical, it was seen in IPA and at first mitigated (retries) and then switched to dbus.

Reply:
That's a very strong point. In this case using dbus for quering the information and the cli for requesting the certs would mitigate that?